polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -224,6 +224,7 @@ impl PeerData {
 		&mut self,
 		on_relay_parent: Hash,
 		relay_parent_mode: ProspectiveParachainsMode,
+		use_non_prospective_chain_mode: bool,
```

Comment:
This parameter is unnecessary and can be derived from `relay_parent_mode` and the peer's CollationVersion.

Reply:
Ahhhh! Good point. I should not program while doing a call :P